flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -990,7 +1072,7 @@ public void serialize(NFA<T> record, DataOutputView target) throws IOException {
 			boolean handleTimeout = source.readBoolean();
 
 			NFA<T> nfa = new NFA<>(eventSerializer, windowTime, handleTimeout);
-			nfa.states = states;
+			nfa.addStates(states);
```

Comment:
This change can result in multiple start states being added. Anyway this is unrelated change